docs/csharp/tutorials/nullable-reference-types.md

```diff
@@ -0,0 +1,225 @@
+    }
+}
+```
+
+As before, you must initialize the list object to a non-null value or the compiler issues a warning. There's no null checks in the second overload of `AddQuestion` because it's not needed: You've declared that variable to be non-nullable. Its value can't be `null`.
+
+Switch to `program.cs` in your editor, and replace the contents of `Main` with the following lines of code:
+
+[!code-csharp[AddQuestions](../../../samples/csharp/NullableIntroduction/NullableIntroduction/Program.cs#AddQuestions)]
+
+Without the `#nullable enable` pragma at the top of the file, the compiler doesn't issue a warning when you pass `null` as the text for the `AddQuestion` argument. Fix that by adding `#nullable enable` following the `using` statement. Now, that last line generates a warning. Remove that line because it can cause bugs later.
+
+## Create respondents and get answers to the survey
+
+Next, write the code that generates answers to the survey. This involves several small tasks:
+
+1. Build a method that generates respondent objects. These represent people asked to fill out the survey.
+1. Build logic to simulate asking the questions to a respondent, collecting answers or noting that a respondent didn't answer.
+1. Repeat until enough respondents have answered the survey.
+
+You'll need a class to represent a survey response, so add that now. Enable nullable support. Add an Id field and a constructor that initializes the ID field, as shown in the following code:
+
+```csharp
+#nullable enable
+namespace NullableIntroduction
+{
+    public class SurveyResponse
+    {
+        public int Id { get; }
+
+        public SurveyResponse(int id) => Id = id;
+    }
+}
+```
+
+Next, add a `static` method to create new participants by generating a random id:
+
+[!code-csharp[GenerateRespondents](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs#Random)]
+
+The main responsibility of this class is to generate the responses for a participant to the questions in the survey. This responsibility has a few steps:
+
+1. Ask for participation in the survey. If the person doesn't consent, return a missing (or null) response.
+1. Ask each question and record the answer. Each answer may also be missing (or null).
+
+Add the following code to your `SurveyRespondent` class:
+
+[!code-csharp[AnswerSurvey](../../../samples/csharp/NullableIntroduction/NullableIntroduction/SurveyResponse.cs#AnswerSurvey)]
+
+The storage for the survey answers is a `Dictionary<int, string>?`, indicating that it may be null. You're using the new language feature to declare your design intent, both to the compiler and to anyone reading your code later. If you ever dereference `surveyResponses` without checking for the null value first, you'll get a compiler warning. You don't get a warning in the `AnswerSurvey` method because the compiler can determine the `surveyResponses` variable was set to a non-null value above.
```

Comment:
you sometimes say dereference with a hyphen and sometimes without. Should be consistent.

Reply:
The spec uses "dereference" (no hyphen), so I standardized on that.